src/components/Report.tsx

```diff
@@ -1,15 +1,43 @@
 import React, { useState, useEffect } from "react";
-import { Col, Row, Container } from "reactstrap";
 import gradesData from "../grades.json";
 import gradeScaleData from "../utils/gradeScale.json";
-// import Logo from "../assets/unnamed.jpg";
-import Logo from "../assets/sept-school-logo.png";
+import attendanceData from "../utils/attendance.json";
+import Logo from "../assets/unnamed.jpg";
+import { calculateCredits } from "../utils/creditCalculator";
+import { rateAttendance, AttendanceRecord } from "../utils/attendanceRating";
+import courseDescriptions from "../utils/courseDescriptions";
+import { TOTAL_SESSIONS, MAX_CREDITS } from "../utils/config";
+import "../Report.css";
+
+// Constants for attendance indices
+const ABSENCES = attendanceData.attendanceCategories.indexOf("A"); // Index for 'A' (Absences)
+const LATES = attendanceData.attendanceCategories.indexOf("L"); // Index for 'L' (Lates)
+const LE = attendanceData.attendanceCategories.indexOf("LE"); // Index for 'LE' (Left Early)
+
+// Helper functions to calculate absences and lates
+const getAbsences = (attendance: number[]): number => {
+  return attendance[ABSENCES];
+};
+
+const getLates = (attendance: number[]): number => {
+  return attendance[LATES] + attendance[LE];
+};
 
 interface Student {
   lastName: string;
   firstName: string;
   email: string;
   overallGrade: number | null;
+  attendance?: AttendanceRecord;
+  courseName: string;
+}
+
+interface RawStudent {
+  lastName: string;
+  firstName: string;
+  email: string;
+  overallGrade: string | null;
```

Comment:
Is this really `null` or should this be an optional field like `courseName`? Same consideration for `Student`. In general I use `undefined` (or optional field) to indicate that I do not have that information, and `null` to indicate that I have the info and it is empty. Not everybody might agree with this statement and I am myself inconsistent about it.